src/main/java/com/yelp/nrtsearch/server/handler/AddDocumentHandler.java

```diff
@@ -61,12 +61,12 @@ public AddDocumentHandler(GlobalState globalState) {
   public StreamObserver<AddDocumentRequest> handle(
       StreamObserver<AddDocumentResponse> responseObserver) {
     return new StreamObserver<>() {
-      Multimap<String, Future<Long>> futures = HashMultimap.create();
+      final Multimap<String, Future<Long>> futures = HashMultimap.create();
```

Comment:
Why is it that these variables are made final but some other variables in this same file are not?
I also prefer to avoid making function-scope variables final - I feel like it adds verbosity but not much value in most cases.

Reply:
These are not really function-scope variables, they are class variables for the `StreamObserver` instance.